diff --git a/src/library_gl.js b/src/library_gl.js
index 9919819..c9ba022 100644
--- a/src/library_gl.js
+++ b/src/library_gl.js
@@ -148,16 +148,22 @@ var LibraryGL = {
         case 0x1406 /* GL_FLOAT */:
           assert(GL.floatExt, 'Must have OES_texture_float to use float textures');
           switch (format) {
+            case 0x1906 /* GL_ALPHA */:
+            case 0x1909 /* GL_LUMINANCE */:
+              sizePerPixel = 1*4;
+              break;
             case 0x1907 /* GL_RGB */:
               sizePerPixel = 3*4;
               break;
             case 0x1908 /* GL_RGBA */:
               sizePerPixel = 4*4;
               break;
+            case 0x190A /* GL_LUMINANCE_ALPHA */:
+              sizePerPixel = 2*4;
+              break;
             default:
               throw 'Invalid format (' + format + ')';
           }
-          internalFormat = Module.ctx.RGBA;
           break;
         default:
           throw 'Invalid type (' + type + ')';
